R/clean_glm_data.R

```diff
@@ -0,0 +1,27 @@
+clean_glm_data <- function(mod, data, weights){
+
+  # Grab only necessary variables
+  data <- get_all_vars(mod, data)
+
+  if(!is.null(weights) & length(weights) != nrow(data))
+    stop('`weights` and `data` must be the same length.')
+
+  # Add weights
```

Comment:
If the user goes with the default of `mod$prior.weights`, that vector is already subset to the complete cases used in the model. That means `marg` will actually fail if you just try to give it a glm object and trust that it'll handle the data correctly, if rows were dropped along the way (since `mod$data` has the full dataset). Example:

```
  data(margex)
  margex$age[c(18, 28, 190, 2830)] <- NA
  margex$treatment[c(5, 29, 2)] <- NA
  margex$y[c(1:20)] <- NA
  margex$distance[c(10, 38, 28, 18, 123, 209, 840)] <- NA
  z1 <- marg(mod = mm, var_interest = 'treatment', type = 'levels')[[1]]
```